src/plans/stepsize.jl

```diff
@@ -660,30 +660,604 @@ If ``$(_tex(:norm, "X_k"; index = "p_k")) ≤ αω_{k-1}``, then let
+
+## Keyword arguments
+
+* `adaptive=true`: switches the `gradient_reduction ``α`` (if `true`) to `0`.
+* `alternate_bound = (bk, hat_c) ->  min(gradient_bound == 0 ? 1.0 : gradient_bound, max(minimal_bound, bk / (3 * hat_c))`:
+  how to determine ``$(_tex(:hat, "k"))_k`` as a function of `(bmin, bk, hat_c) -> hat_bk`
+* `count_threshold=4`:  an `Integer` for ``$(_tex(:hat, "c"))``
+* `gradient_reduction::R=adaptive ? 0.9 : 0.0`: the gradient reduction factor threshold ``α ∈ [0,1)``
+* `gradient_bound=norm(M, p, X)`: the bound ``b_k``.
+* `minimal_bound=1e-4`: the value ``b_{$(_tex(:text, "min"))}``
+$(_var(:Keyword, :p; add = "only used to define the `gradient_bound`"))
+$(_var(:Keyword, :X; add = "only used to define the `gradient_bound`"))
+"""
+function AdaptiveWNGradient(args...; kwargs...)
+    return ManifoldDefaultsFactory(Manopt.AdaptiveWNGradientStepsize, args...; kwargs...)
+end
+
+
+@doc """
+    CubicBracketingLinesearchStepsize{P,T,R<:Real} <: Linesearch
+
+Do a bracketing line search to find a step size ``α`` that finds a
+local minimum along the  search direction ``X`` starting from ``p``,
+utilizing cubic polynomial interpolation.
+See [`CubicBracketingLinesearch`](@ref) for the mathematical details.
+
+# Fields
+$(_var(:Field, :p, "candidate_point"; add = " as temporary storage for candidates"))
+* `initial_stepsize::R`: the step size to start the search with
+* `last_stepsize::R`
+$(_var(:Field, :retraction_method))
+* `stepsize_increase::R`:  step size increase factor ``>1``
+* `max_iterations::I`: maximum number of iterations
+* `sufficient_curvature::R`: target reduction of the curvature ``(0,1)``
+* `min_bracket_width::R`: minimal size of the bracket ``[a,b]``
+* `hybrid::Bool`: use the hybrid strategy
+$(_var(:Field, :vector_transport_method))
+
+# Constructor
+
+    CubicBracketingLinesearchStepsize(M::AbstractManifold; kwargs...)
+
+## Keyword arguments
+
+$(_var(:Keyword, :p, "candidate_point"; add = " as temporary storage for candidates"))
+* `initial_stepsize=1.0`: the step size to start the search with
+$(_var(:Keyword, :retraction_method))
+* `stepsize_increase=1.1`:  step size increase factor ``>1``
+* `max_iterations=100`: maximum number of iterations
+* `sufficient_curvature=0.2`: target reduction of the curvature ``(0,1)``
+* `min_bracket_width=1e-4`: minimal size of the bracket ``[a,b]``
+* `hybrid=true`: use the hybrid strategy
+$(_var(:Keyword, :vector_transport_method))
+"""
+mutable struct CubicBracketingLinesearchStepsize{
+        P,
+        R <: Real,
+        I <: Integer,
+        TRM <: AbstractRetractionMethod,
+        VTM <: AbstractVectorTransportMethod,
+    } <: Linesearch
+    candidate_point::P
+    initial_stepsize::R
+    last_stepsize::R
+    retraction_method::TRM
+    stepsize_increase::R
+    max_iterations::I
+    sufficient_curvature::R
+    min_bracket_width::R
+    hybrid::Bool
+    vector_transport_method::VTM
+    function CubicBracketingLinesearchStepsize(
```

Comment:
I'd suggest adding `max_stepsize` parameter here (an upper bound on the stepsize), initialized by `max_stepsize(M, candidate_point)`.